core/src/main/java/org/apache/iceberg/TableProperties.java

```diff
@@ -91,6 +91,8 @@ private TableProperties() {
 
   public static final String WRITE_LOCATION_PROVIDER_IMPL = "write.location-provider.impl";
 
+  public static final String WRITE_FILE_IO_IMPL = "write.file-io.impl";
```

Comment:
This isn't just a write option. It controls how files are read as well.

How about using an `io` namespace instead? Maybe `io.file-io.impl` or `io.impl`?

Reply:
Yeah I used write because there is no namespace for both read and write. I can change to `io.file-io.impl`, `io.impl` seems too generic in case we want to have something else in the same namespace in the future.